src/checkoutservice/genproto/demo.pb.go

```diff
@@ -13,850 +13,1016 @@
-	return xxx_messageInfo_ListRecommendationsResponse.Marshal(b, m, deterministic)
-}
-func (m *ListRecommendationsResponse) XXX_Merge(src proto.Message) {
-	xxx_messageInfo_ListRecommendationsResponse.Merge(m, src)
+func (x *ListRecommendationsResponse) Reset() {
+	*x = ListRecommendationsResponse{}
+	if protoimpl.UnsafeEnabled {
+		mi := &file_demo_proto_msgTypes[7]
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		ms.StoreMessageInfo(mi)
+	}
 }
-func (m *ListRecommendationsResponse) XXX_Size() int {
-	return xxx_messageInfo_ListRecommendationsResponse.Size(m)
+
+func (x *ListRecommendationsResponse) String() string {
+	return protoimpl.X.MessageStringOf(x)
 }
-func (m *ListRecommendationsResponse) XXX_DiscardUnknown() {
-	xxx_messageInfo_ListRecommendationsResponse.DiscardUnknown(m)
+
+func (*ListRecommendationsResponse) ProtoMessage() {}
+
+func (x *ListRecommendationsResponse) ProtoReflect() protoreflect.Message {
+	mi := &file_demo_proto_msgTypes[7]
+	if protoimpl.UnsafeEnabled && x != nil {
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		if ms.LoadMessageInfo() == nil {
+			ms.StoreMessageInfo(mi)
+		}
+		return ms
+	}
+	return mi.MessageOf(x)
 }
 
-var xxx_messageInfo_ListRecommendationsResponse proto.InternalMessageInfo
+// Deprecated: Use ListRecommendationsResponse.ProtoReflect.Descriptor instead.
+func (*ListRecommendationsResponse) Descriptor() ([]byte, []int) {
+	return file_demo_proto_rawDescGZIP(), []int{7}
+}
 
-func (m *ListRecommendationsResponse) GetProductIds() []string {
-	if m != nil {
-		return m.ProductIds
+func (x *ListRecommendationsResponse) GetProductIds() []string {
+	if x != nil {
+		return x.ProductIds
 	}
 	return nil
 }
 
 type Product struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
 	Id          string `protobuf:"bytes,1,opt,name=id,proto3" json:"id,omitempty"`
 	Name        string `protobuf:"bytes,2,opt,name=name,proto3" json:"name,omitempty"`
 	Description string `protobuf:"bytes,3,opt,name=description,proto3" json:"description,omitempty"`
 	Picture     string `protobuf:"bytes,4,opt,name=picture,proto3" json:"picture,omitempty"`
 	PriceUsd    *Money `protobuf:"bytes,5,opt,name=price_usd,json=priceUsd,proto3" json:"price_usd,omitempty"`
 	// Categories such as "vintage" or "gardening" that can be used to look up
 	// other related products.
-	Categories           []string `protobuf:"bytes,6,rep,name=categories,proto3" json:"categories,omitempty"`
-	XXX_NoUnkeyedLiteral struct{} `json:"-"`
-	XXX_unrecognized     []byte   `json:"-"`
-	XXX_sizecache        int32    `json:"-"`
+	Categories []string `protobuf:"bytes,6,rep,name=categories,proto3" json:"categories,omitempty"`
+	Discount   int64    `protobuf:"varint,7,opt,name=discount,proto3" json:"discount,omitempty"`
+	Savings 	int64    `protobuf:"varint,8,opt,name=savings,proto3" json:"savings,omitempty"`
```

Comment:
Unfortunately the `*.pb.go` files are generated by `protoc`. If we touch it manually, it will be overwritten next time we have to touch the GRPC declaration :/

Could we move the savings somewhere else? Or calculate the Savings from the Price * Discount?